erna/automatic_processing/qsub.py

```diff
@@ -0,0 +1,199 @@
+import xmltodict
+import subprocess as sp
+import os
+import pandas as pd
+import logging
+
+from .utils import get_aux_dir
+from .database_utils import (
+    build_output_base_name, build_output_directory_name,
+    save_xml, save_jar
+)
+from .database import ProcessingState
+
+log = logging.getLogger(__name__)
+
+
+def get_current_jobs(user=None):
+    ''' Return a dataframe with current jobs of user '''
+    user = user or os.environ['USER']
+    xml = sp.check_output(['qstat', '-u', user, '-xml']).decode()
+    data = xmltodict.parse(xml)
+    job_info = data['job_info']
+    queue_info = job_info['queue_info']
+    job_info = job_info['job_info']
+    queued_jobs = queue_info['job_list'] if queue_info else []
+    running_jobs = job_info['job_list'] if job_info else []
+
+    df = pd.DataFrame(columns=[
+        '@state', 'JB_job_number', 'JAT_prio', 'JB_name', 'JB_owner',
+        'state', 'JB_submission_time', 'queue_name', 'slots', 'JAT_start_time'
+    ])
+
+    if not isinstance(running_jobs, list):
+        running_jobs = [running_jobs]
+    if not isinstance(queued_jobs, list):
+        queued_jobs = [queued_jobs]
+
+    df = df.append(pd.DataFrame(running_jobs + queued_jobs), ignore_index=True)
+
+    if len(df) == 0:
+        return df
+
+    df.drop('state', axis=1, inplace=True)
+    df.rename(inplace=True, columns={
+        '@state': 'state',
+        'JB_owner': 'owner',
+        'JB_name': 'name',
+        'JB_job_number': 'job_number',
+        'JB_submission_time': 'submission_time',
+        'JAT_prio': 'priority',
+        'JAT_start_time': 'start_time',
+    })
+
+    df = df.astype({'job_number': int, 'priority': float})
+    df['start_time'] = pd.to_datetime(df['start_time'])
+    df['submission_time'] = pd.to_datetime(df['submission_time'])
+    return df
+
+
+def build_qsub_command(
+        executable,
+        stdout=None,
+        stderr=None,
+        job_name=None,
+        queue=None,
+        mail_address=None,
+        mail_settings='a',
+        environment=None,
+        resources=None,
+        engine='SGE',
+        ):
+    command = []
+    command.append('qsub')
+
+    if job_name:
+        command.extend(['-N', job_name])
+
+    if queue:
+        command.extend(['-q', queue])
+
+    if mail_address:
+        command.extend(['-M', mail_address])
+
+    command.extend(['-m', mail_settings])
+
+    # allow a binary executable
+    if engine == 'SGE':
```

Comment:
The default is in `qsub`. The parameter is not necessary if not on SGE, torque permits non-bashscript executables by default.